ynh_check_starting/ynh_check_starting

```diff
@@ -2,46 +2,64 @@
 
 # Start or restart a service and follow its booting
 #
-# usage: ynh_check_starting "Line to match" [Log file] [Timeout]
+# usage: ynh_check_starting "Line to match" [service name] [Log file] [Timeout]
 #
 # | arg: Line to match - The line to find in the log to attest the service have finished to boot.
 # | arg: Log file - The log file to watch
 # /var/log/$app/$app.log will be used if no other log is defined.
 # | arg: Timeout - The maximum time to wait before ending the watching. Defaut 300 seconds.
 ynh_check_starting () {
 	local line_to_match="$1"
-	local app_log="${2:-/var/log/$app/$app.log}"
-	local timeout=${3:-300}
+	local service_name="${2:-$app}"
+	local app_log="${3:-/var/log/$app/$app.log}"
+	local timeout=${4:-300}
 
 	ynh_clean_check_starting () {
 		# Stop the execution of tail.
 		kill -s 15 $pid_tail 2>&1
 		ynh_secure_remove "$templog" 2>&1
 	}
 
-	echo "Starting of $app" >&2
-	systemctl restart $app
+	echo "Starting of $service_name" >&2
+	systemctl restart $service_name
+	
+	local i=0
 	local templog="$(mktemp)"
+	
+	# Wait until the log file exit
+	if [[ ! -e $app_log ]]
+	then
+		for i in $(seq 1 $timeout)
+		do
+			if [[ -e $app_log ]]
+			then
+				cat $app_log > "$templog"
+				break
+			fi
+			echo -n "." >&2
+			sleep 1
+		done
+	fi
+	
 	# Following the starting of the app in its log
-	tail -f -n1 "$app_log" > "$templog" &
+	tail -f -n1 "$app_log" >> "$templog" &
```

Comment:
Instead of a loop waiting for the existence of the log file, it would be really simpler to use '-F' instead of '-f' for this tail.